code/game/machinery/ARES/ARES_interface_admin.dm

```diff
@@ -0,0 +1,494 @@
+				lock_status = TICKET_CLOSED
+
+		var/list/current_maint = list()
+		current_maint["id"] = maint_ticket.ticket_id
+		current_maint["time"] = maint_ticket.ticket_time
+		current_maint["priority_status"] = maint_ticket.ticket_priority
+		current_maint["category"] = maint_ticket.ticket_name
+		current_maint["details"] = maint_ticket.ticket_details
+		current_maint["status"] = maint_ticket.ticket_status
+		current_maint["submitter"] = maint_ticket.ticket_submitter
+		current_maint["assignee"] = maint_ticket.ticket_assignee
+		current_maint["lock_status"] = lock_status
+		current_maint["ref"] = "\ref[maint_ticket]"
+		logged_maintenance += list(current_maint)
+	data["maintenance_tickets"] = logged_maintenance
+
+	var/list/logged_access = list()
+	for(var/datum/ares_ticket/access/access_ticket as anything in tickets_access)
+		var/lock_status = TICKET_OPEN
+		switch(access_ticket.ticket_status)
+			if(TICKET_REJECTED, TICKET_CANCELLED, TICKET_REVOKED)
+				lock_status = TICKET_CLOSED
+
+		var/list/current_ticket = list()
+		current_ticket["id"] = access_ticket.ticket_id
+		current_ticket["time"] = access_ticket.ticket_time
+		current_ticket["priority_status"] = access_ticket.ticket_priority
+		current_ticket["title"] = access_ticket.ticket_name
+		current_ticket["details"] = access_ticket.ticket_details
+		current_ticket["status"] = access_ticket.ticket_status
+		current_ticket["submitter"] = access_ticket.ticket_submitter
+		current_ticket["assignee"] = access_ticket.ticket_assignee
+		current_ticket["lock_status"] = lock_status
+		current_ticket["ref"] = "\ref[access_ticket]"
+		logged_access += list(current_ticket)
+	data["access_tickets"] = logged_access
+
+	return data
+
+
+/datum/ares_link/ui_state(mob/user)
+	return GLOB.admin_state
+
+/datum/ares_link/ui_close(mob/user)
+	. = ..()
+	if(admin_interface.logged_in && (user.ckey == admin_interface.logged_in))
+		admin_interface.current_menu = "login"
+		admin_interface.last_menu = "login"
+		admin_interface.access_list += "[admin_interface.logged_in] logged out at [worldtime2text()]."
+		admin_interface.last_login = admin_interface.logged_in
+		admin_interface.logged_in = null
+
+/datum/ares_link/ui_act(action, list/params, datum/tgui/ui, datum/ui_state/state)
+	. = ..()
+	if(.)
+		return
+	var/mob/user = usr
+
+	switch (action)
+		if("go_back")
+			if(!admin_interface.last_menu)
+				return to_chat(user, SPAN_WARNING("Error, no previous page detected."))
+			var/temp_holder = admin_interface.current_menu
+			admin_interface.current_menu = admin_interface.last_menu
+			admin_interface.last_menu = temp_holder
+
+		if("login")
+			if(check_rights_for(user.client, R_MOD))
+				admin_interface.logged_in = user.client.ckey
+				admin_interface.access_list += "[user.client.ckey] at [worldtime2text()], Access Level '[user.client.admin_holder?.rank]'."
+			else
+				to_chat(user, SPAN_WARNING("You require staff identification to access this terminal!"))
+				return FALSE
```

Comment:
Either this needs to be done for every ui_act action that assumes login has occurred; or every ui_act action that assumes login has occured needs to verify the `admin_interface.logged_in` is the user interacting with this